### PR TITLE
feat: Automate Contract Verification with Hardhat

### PR DESCRIPTION
**Pull request by gr8h**
This PR introduces a new feature for automated verification of smart contracts using the Hardhat framework.

### Technical Details
* A new Hardhat script `4_verify_contracts.js` has been added to handle the verification of contracts.
* Updated hardhat.config.js to include necessary configurations for custom chains fuse & spark

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -41,6 +42,12 @@ module.exports = {
       accounts: getSigners(),
       allowUnlimitedContractSize: true,
     },
+    shadowfork: {
+      url: "http://35.241.160.82:8545",
```
Comment (leonprou) [inline hardhat.config.js:46]: The URL should be moved to the config to we can delete the shawork completely IMO

File: scripts/4_transfer_balance.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+require("dotenv").config();
+const hre = require("hardhat");
+const ethers = hre.ethers;
+const { assert } = require("chai");
+
+const { CREDENTIALS_ADDRESS } = process.env;
+
+async function main() {
```
Comment (leonprou) [inline scripts/4_transfer_balance.js:8]: @gr8h what this script is doing?

**Review by leonprou — COMMENTED** [review]
LGTM
just left some comments

File: hardhat.config.js (hunk 1 above)
Comment (gr8h) [reply hardhat.config.js]: Done

File: scripts/4_transfer_balance.js (hunk 2 above)
Comment (gr8h) [reply scripts/4_transfer_balance.js]: @leonprou  removed was just temporary added to transfer some funds

**Review by leonprou — APPROVED** [review]
LGTM,
thanks

**Pull request merged**